pkg/storage/sqlite/sqlite.go

```diff
@@ -83,6 +83,91 @@ func (s SQLite) ListIndexers(ctx context.Context) ([]*model.Indexers, error) {
 	return indexers, nil
 }
 
+// CreateMovie stores a movie
+func (s SQLite) CreateMovie(ctx context.Context, movie model.Movies) (int32, error) {
+	// log := logger.FromCtx(ctx)
+
+	stmt := table.Movies.INSERT(table.Movies.MutableColumns).RETURNING(table.Movies.ID).MODEL(movie).ON_CONFLICT(table.Movies.ID).DO_NOTHING()
+	result, err := s.handleInsert(ctx, stmt)
+	if err != nil {
+		return 0, err
+	}
+
+	inserted, err := result.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+	// hope for the best I guess
+	res := int32(inserted)
+	return res, nil
+}
+
+// DeleteMovie removes a movie by id
+func (s SQLite) DeleteMovie(ctx context.Context, id int64) error {
+	stmt := table.Movies.DELETE().WHERE(table.Movies.ID.EQ(sqlite.Int64(id))).RETURNING(table.Movies.ID)
+	_, err := s.handleDelete(ctx, stmt)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ListMovies lists the stored movies
+func (s SQLite) ListMovies(ctx context.Context) ([]*model.Movies, error) {
+	log := logger.FromCtx(ctx)
+
+	movies := make([]*model.Movies, 0)
+	stmt := table.Movies.SELECT(table.Movies.AllColumns).FROM(table.Movies).ORDER_BY(table.Movies.Added.ASC())
+	err := stmt.QueryContext(ctx, s.db, &movies)
+	if err != nil {
+		log.Errorf("failed to list indexer: %w", err)
+	}
+
+	return movies, nil
+}
+
+// CreateMovieFile stores a movie file
+func (s SQLite) CreateMovieFile(ctx context.Context, file model.MovieFiles) (int32, error) {
+	// Exclude DateAdded so that the default is used
+	stmt := table.MovieFiles.INSERT(table.MovieFiles.MutableColumns.Except(table.MovieFiles.DateAdded)).RETURNING(table.MovieFiles.ID).MODEL(file)
+	result, err := s.handleInsert(ctx, stmt)
+	if err != nil {
+		return 0, err
+	}
+
+	inserted, err := result.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+	// hope for the best I guess
```

Comment:
I was trying to remember why I was returning int32 in other funcs but I think it was because the generated code used int32 for the id fields.. we should prob just set this as int64 on the generated structs or find a way to generate code that way

I did see this.. https://github.com/go-jet/jet/blob/f2e4b8551c48b97d0cd2d3deff47dc1b2aa2f04e/tests/postgres/generator_template_test.go#L359

Reply:
Actually I think I misread what this was doing